R/imports.R

```diff
@@ -1,4 +1,14 @@
-#' @importFrom rlang %||% abort warn .data .env
+#' @importFrom vctrs vec_assert vec_set_names
```

Comment:
I do not fully understand the alternatives that are being suggested

Reply:
As a maintainer I believe you're welcome to define your preference and I'm happy to follow. I take this as an opportunity to explain my own preferences, which you'll see baked into most r2dii packages on CRAN.

1. The namespace of the package is insulated from the global environment. If the user defines a function with the same name (e.g. `abort <- function() stop("Oh no!")`) the functions in this package will ignore it and still find the one that comes from `#' @importFrom pkg fun`.

2. Great you're thinking on a package-wide convention. Once you define those conventions, I'm more than happy to follow. I don't usually need to define any of my preferences other than saying something like "I use style, patterns, and conventions from the tidyverse team".

Reply:
@maurolepore  I find aspect 1. very convincing, so let us follow this convention
@jacobvjk There is basically 3 ways to make functions available

1. using :: so dplyr::mutate
2. defining all imports in imports (mauros suggestion)
3. defining imports not in imports.R but in function definitions

I would be against 3, since it calls for clutter and inconsistency, but am fine with 2. :)